### PR TITLE
Refactor Kryo into sub module just like AMQP

### PR DESCRIPTION
**Pull request by fenryka**
Prior to enabling AMQP RPC move Kryo code to it's own submodule just like AMQP

### CODE REVIEW

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt [hunk 1]
```diff
@@ -77,6 +78,7 @@ class CordaRPCClient @JvmOverloads constructor(
     private val rpcClient = RPCClient<CordaRPCOps>(
             tcpTransport(ConnectionDirection.Outbound(), hostAndPort, config = null),
             configuration.toRpcClientConfiguration(),
+            //AMQP_RPC_CLIENT_CONTEXT
```
Comment (andr3ej) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: commented code shouldn't really go into master

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoSerialisationScheme.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
+package net.corda.nodeapi.internal.serialization.kryo
```
Comment (andr3ej) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoSerialisationScheme.kt]: british english is fine in comments and docs but this filename should be US english for consistency with others

File: node/src/main/kotlin/net/corda/node/internal/Node.kt [hunk 3]
```diff
@@ -370,6 +371,11 @@ open class Node(configuration: FullNodeConfiguration,
             registerScheme(KryoServerSerializationScheme())
             registerScheme(AMQPServerSerializationScheme())
         }
+        /*
+        SerializationDefaults.P2P_CONTEXT = AMQP_P2P_CONTEXT.withClassLoader(classloader)
+        SerializationDefaults.RPC_SERVER_CONTEXT = AMQP_RPC_SERVER_CONTEXT.withClassLoader(classloader)
+        SerializationDefaults.STORAGE_CONTEXT = AMQP_STORAGE_CONTEXT.withClassLoader(classloader)
+        */
```
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/internal/Node.kt]: if this is effectively a todo, i don't think it's appropriate to use master for that

Comment (fenryka) [reply node/src/main/kotlin/net/corda/node/internal/Node.kt]: You haven't seem master recently, it's full of TODO's,

Comment (mnesbit) [reply node/src/main/kotlin/net/corda/node/internal/Node.kt]: I agree I don't like commented out code either. If we need a config switch lets make it one.

Comment (andr3ej) [reply node/src/main/kotlin/net/corda/node/internal/Node.kt]: @fenryka master is probably not the best place to learn best practices

**Comment by fenryka** [comment]
So the commented code is really only there because it's all going to change in the next commit given this is just a prelim to actually enabling AMQP, so meh, but people seem super keen on them going away now and not soon so I'll remove it

File: node/src/main/kotlin/net/corda/node/internal/Node.kt (hunk 3 above)
Comment (fenryka) [reply node/src/main/kotlin/net/corda/node/internal/Node.kt]: I learned those 15 years ago ;)

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoSerialisationScheme.kt (hunk 2 above)
Comment (fenryka) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoSerialisationScheme.kt]: that is a good point!

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt (hunk 1 above)
Comment (fenryka) [reply client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: done

**Comment by fenryka** [comment]
@mnesbit It's kinda the way it was done "at the time", ultimately given we're going to rip out Kryo except for storage the lack of matrix aplicability won't be too noticable

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt [hunk 4]
```diff
@@ -12,6 +12,7 @@ import net.corda.core.serialization.CordaSerializable
 import net.corda.core.serialization.SerializationContext
 import net.corda.core.utilities.loggerFor
 import net.corda.nodeapi.internal.serialization.amqp.hasAnnotationInHierarchy
+import net.corda.nodeapi.internal.serialization.kryo.ThrowableSerializer
```
Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: Is this spurious?

Comment (fenryka) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: it is not :)

When ThrowableSEializer was in the higher level serialisation package we could just resolve it, now it's moved to a sub module we need to explicitly import it

Comment (rick-r3) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/CordaClassResolver.kt]: Makes sense

**Pull request merged**